script/DeployYieldStreamingFactory.s.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.19;
+
+import {MULTISIG} from "./base/CommonAddresses.sol";
+import {DeployScriptBase} from "./base/DeployScriptBase.sol";
+import {YieldStreamingFactory} from "../src/YieldStreamingFactory.sol";
+
+contract DeployStreamHubFactory is DeployScriptBase {
+    address instanceOwner;
+
+    function _initStateVars() internal override {
+        super._initStateVars();
+
+        instanceOwner = MULTISIG;
+    }
+
+    function run() external virtual returns (YieldStreamingFactory streamHubFactory) {
+        vm.startBroadcast(deployer);
+
+        streamHubFactory = YieldStreamingFactory(
+            create3.deploy(
```

Comment:
does this mean we get the factory on the same contract address on all chains its deployed on?

Reply:
it should and I can check it to confirm. We would have to update the address for `CREATE3Factory` inside the `CREATE3Script`

Reply:
just confirmed that it works as expected